### PR TITLE
COCICCS-5814,COCICCS-5693

### PR DESCRIPTION
**Pull request by nithyal1**

### CODE REVIEW

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java [hunk 1]
```diff
@@ -9894,4 +9947,290 @@ public Response udapteCertMetadataOnAppliationChange(String token, String certPa
...
+			
+						
+			
+			return isCertDataUpdated;
+		}
+    
+    
+    
+    
+	 private boolean checkAppDetailsAvailable(String appName, String token) {
```
Comment (alexrajan3) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java:10021]: Add method description

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java [hunk 2]
```diff
@@ -9894,4 +9947,290 @@ public Response udapteCertMetadataOnAppliationChange(String token, String certPa
...
+			if (!HttpStatus.OK.equals(response.getHttpstatus())) {
+				isDataAvailable = false;
+			}
+			return isDataAvailable;
+	 }
+	 
+		 
+	 
+	 
+	 private TMOAppMetadataDetails addCertToAppList(TMOAppMetadataDetails details,String appName, String certType, String certName, String token) {
```
Comment (alexrajan3) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java:10049]: Add method description

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java [hunk 3]
```diff
@@ -9894,4 +9947,290 @@ public Response udapteCertMetadataOnAppliationChange(String token, String certPa
...
+				 details.setExternalCertificateList(certListOld);
+			 }else {
+				 details.setInternalCertificateList(certListOld);
+			 }
+			 }
+			return details;
+	 }
+	 
+	 
+	 private boolean updatecertificateMetadataForApplicationDetails(JsonObject object, String token) {
```
Comment (alexrajan3) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java:10111]: Add method description

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java [hunk 4]
```diff
@@ -9894,4 +9947,290 @@ public Response udapteCertMetadataOnAppliationChange(String token, String certPa
...
+						put(LogMessage.APIURL, ThreadLocalContext.getCurrentMap().get(LogMessage.APIURL)).
+						build()));
+			}
+			
+				}			
+			
+			return isCertDataUpdated;
+		}
+	 
+	 private TMOAppMetadataDetails deleteCertFromAppList(TMOAppMetadataDetails details,String appName, String certType, String certName, String token) {
```
Comment (alexrajan3) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java:10159]: Add method description

**Review by alexrajan3 — CHANGES_REQUESTED** [review]
Check the review comments

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/common/SSLCertificateConstants.java [hunk 5]
```diff
@@ -136,4 +136,5 @@ private SSLCertificateConstants() {}
     public static final String CERTIFICATE_NAME="certificateName";
     public static final String GET_PENDING_APPROVAL_CERT_PROCESS="getPendApprovalCertsAndProcess";
     public static final String STATUS_WAITING = "waiting";
+	public static final String APPLICATION_DETAILS_PATH = "tmo-applications/";
```
Comment (sarathmohan4) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/common/SSLCertificateConstants.java:139]: is this variable used?

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java [hunk 6]
```diff
@@ -6732,6 +6748,15 @@ else if (TVaultConstants.OIDC.equals(vaultAuthMethod)) {
 
             JsonParser jsonParser = new JsonParser();
             JsonObject object = ((JsonObject) jsonParser.parse(response.getResponse())).getAsJsonObject("data");
+            
+            //remove certificate name from application metadata
+            boolean appMetadataStatus = updatecertificateMetadataForApplicationDetails(object, authToken);
```
Comment (sarathmohan4) [inline tvaultapi/src/main/java/com/tmobile/cso/vault/api/service/SSLCertificateService.java:6753]: can we add logs based on the update status or if is it is done in the function itself?

**Review by sarathmohan4 — CHANGES_REQUESTED** [review]

File: tvaultapi/src/main/java/com/tmobile/cso/vault/api/common/SSLCertificateConstants.java (hunk 5 above)
Comment (nithyal1) [reply tvaultapi/src/main/java/com/tmobile/cso/vault/api/common/SSLCertificateConstants.java]: yes

**Review by alexrajan3 — APPROVED** [review]

**Pull request merged**